internal/provider/resource_cloudflare_list.go

```diff
@@ -209,8 +209,8 @@ func buildListItemsCreateRequest(resource *schema.ResourceData, items []interfac
 			var preservePathSuffix *bool = nil
 
 			hasField := func(field string) bool {
-				_, has := resource.GetOkExists(fmt.Sprintf("item.%d.value.0.redirect.0.%s", i, field))
-				return has
+				v := getRawValue(fmt.Sprintf("item.%d.value.0.redirect.0.%s", i, field), resource.GetRawConfig())
```

Comment:
we really shouldn't be using `getRawValue` as it calls Terraform internal methods without going via the abstraction layers. it was initially planned to help with tristate booleans in Terraform configs but we instead took another approach due to the stability and on going issues with it. it really should have been removed.

what is the issue we're hitting here that requires it?

Reply:
this is the exact issue type we ended up solving in #1689 with a breaking change. as i suspect it is the same underlying cause of tristate boolean values, here was the testing outcome i did then with `d.GetRawConfig`.

|Initial state| Updated state|Outcome|
|---|---|---|
|enabled: false|enabled: true| ✅ |
| enabled: false | enabled: nil | ❌ |
| enabled: true | enabled: false | ✅ |
| enabled: true | enabled: nil | ✅ |
| enabled: nil | enabled: true | ✅ |
| enabled: nil | enabled: false | ❌ |

as you can see, this didn't fully solve the issue despite getting us most of the way there. another drawback is that this approach is using internal methods to Terraform and isn't recommended to be relied upon.

i'll grab some time with you to discuss what we can do going forward here.

Reply:
:( It really shouldn't be this hard to get optional values to work.  In any case, I've updated the PR and avoided the issue by negating the field name, so the default is now `false`.  This is, of course, a breaking change, but the field was misbehaving any way, and was just introduced in the last release.

Reply:
agree; i've flicked you an internal ticket that covers the context and our potential next steps. as this is an issue in Terraform core (not the provider) we are pretty limited in the options to take. 

eventually, we'll move to terraform-plugin-framework which doesn't have this issue with uninitialised boolean values.